### ISSUE TITLE
pm2 monit previous logs gets cleared when navigating to a different process

### ISSUE THREAD

**Issue by aprilmintacpineda** [Unitech/pm2#4565]
## What's going wrong?

I have 2 nodejs processes running. Both do console logs. When I navigate to a different process, the log box gets cleared and when I navigate back to the previous one, the previous logs are lost.

## How could we reproduce this issue?

1. Run 2 or more nodejs processes that does console logs.
2. Go to `pm2 monit` to see the processes.
  - See the 2 processes on the processes box, which is correct.
3. Notice the process you are currently in, let's call that `PROC1`. Also notice that `PROC1` has logs. Using the arrow keys, navigate to a different process, let's call that `PROC2`.
4. (issue 1) The log box will get cleared out and the logs of `PROC2` doesn't show.
5. Using the arrow keys, navigate back to `PROC1`.
6. (issue 2) The log box will get cleared out and the logs of `PROC1` doesn't show.

**issue 1 expected result**

Should have displayed all logs of `PROC2` (could be limited depending on a particular configuration with reasonable default, like maximum log lines).

**issue 2 expected result**

Should have displayed all logs of `PROC1` (could be limited depending on a particular configuration with reasonable default, like maximum log lines).

Thanks you.

```
--- PM2 report ----------------------------------------------------------------
Date                 : Fri Jan 03 2020 14:19:25 GMT+0800 (Philippine Standard Time)
===============================================================================
--- Daemon -------------------------------------------------
pm2d version         : 4.2.1
node version         : 13.5.0
node path            : /usr/local/bin/pm2
argv                 : /usr/local/Cellar/node/13.5.0/bin/node,/usr/local/lib/node_modules/pm2/lib/Daemon.js
argv0                : node
user                 : aprilmintacpineda
uid                  : 501
gid                  : 20
uptime               : 281min
===============================================================================
--- CLI ----------------------------------------------------
local pm2            : 4.2.1
node version         : 13.5.0
node path            : /usr/local/bin/pm2
argv                 : /usr/local/Cellar/node/13.5.0/bin/node,/usr/local/bin/pm2,report
argv0                : node
user                 : aprilmintacpineda
uid                  : 501
gid                  : 20
===============================================================================
--- System info --------------------------------------------
arch                 : x64
platform             : darwin
type                 : Darwin
cpus                 : Intel(R) Core(TM) i7-2620M CPU @ 2.70GHz
cpus nb              : 4
freemem              : 3444666368
totalmem             : 17179869184
home                 : /Users/aprilmintacpineda
===============================================================================
--- PM2 list -----------------------------------------------
┌─────┬──────────────────────┬─────────────┬─────────┬─────────┬──────────┬────────┬──────┬───────────┬──────────┬──────────┬──────────┬──────────┐
│ id  │ name                 │ namespace   │ version │ mode    │ pid      │ uptime │ ↺    │ status    │ cpu      │ mem      │ user     │ watching │
├─────┼──────────────────────┼─────────────┼─────────┼─────────┼──────────┼────────┼──────┼───────────┼──────────┼──────────┼──────────┼──────────┤
│ 2   │ portal-be-cluster    │ default     │ N/A     │ cluster │ 4475     │ 4h     │ 0    │ online    │ 0%       │ 38.4mb   │ apr… │ disabled │
│ 3   │ portal-be-cluster    │ default     │ N/A     │ cluster │ 4476     │ 4h     │ 0    │ online    │ 0%       │ 37.3mb   │ apr… │ disabled │
│ 4   │ portal-be-cluster    │ default     │ N/A     │ cluster │ 4477     │ 4h     │ 0    │ online    │ 0%       │ 37.6mb   │ apr… │ disabled │
│ 5   │ portal-be-cluster    │ default     │ N/A     │ cluster │ 4478     │ 4h     │ 0    │ online    │ 0%       │ 37.1mb   │ apr… │ disabled │
│ 0   │ portal-fe            │ default     │ N/A     │ fork    │ 15555    │ 26m    │ 2    │ online    │ 0%       │ 33.5mb   │ apr… │ disabled │
└─────┴──────────────────────┴─────────────┴─────────┴─────────┴──────────┴────────┴──────┴───────────┴──────────┴──────────┴──────────┴──────────┘
===============================================================================
--- Daemon logs --------------------------------------------
/Users/aprilmintacpineda/.pm2/pm2.log last 20 lines:
PM2        |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
PM2        |     at Object.<anonymous> (/Users/aprilmintacpineda/my-app/node_modules/@babel/node/lib/_babel-node.js:238:23)
PM2        |     at Module._compile (internal/modules/cjs/loader.js:1139:30)
PM2        |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1159:10)
PM2        |     at Module.load (internal/modules/cjs/loader.js:988:32)
PM2        |     at Function.Module._load (internal/modules/cjs/loader.js:896:14)
PM2        |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
PM2        |     at internal/main/run_main_module.js:17:47
PM2        | Emitted 'error' event on WebSocketServer instance at:
PM2        |     at Server.emit (events.js:305:20)
PM2        |     at emitErrorNT (net.js:1335:8)
PM2        |     at processTicksAndRejections (internal/process/task_queues.js:84:21) {
PM2        |   code: 'EADDRINUSE',
PM2        |   errno: -48,
PM2        |   syscall: 'listen',
PM2        |   address: '::',
PM2        |   port: 7071
PM2        | }
PM2        | [nodemon] app crashed - waiting for file changes before starting...
PM2        | [nodemon] app crashed - waiting for file changes before starting...
```

Note that those last 20 lines of the daemon logs are not relevant to this issue. Thanks.

**Comment by bkniffler** [comment]
I don't really get the point of the monit tool with different apps that can be navigated, if each time the log is cleared. If this would work, monit would be such a great tool!

**Comment by aprilmintacpineda** [comment]
Don't you dare!

**Comment by RuRo** [comment]
Come on! @Unitech

**Comment by Organizer21** [comment]
Is this getting a fix?

**Comment by robertgagnon726** [comment]
I don't see the point of logs not persisting while monitoring either. It'd be nice to have an optional config that allows you to persist n number of lines when switching between processes
